flask/app/participants.py

```diff
@@ -37,22 +37,36 @@
 def list_participants():
 
     # parsing query parameters
-    max_rows = request.args.get("max_rows", default=100)
+    limit = request.args.get("limit", default=10)
+    page = request.args.get("page", default=1)
     starts_with = request.args.get("starts_with", default="", type=str)
     starts_with = f"{starts_with}%"  # sql syntax
     order_by_col = request.args.get("order", default="participant_id", type=str)
-    # need some default or we need an ifelse statement, one with order_by method and one without. AFAIK there is no 'default' parameter we can pass into order_by to get the default sql ordering scheme
+    filter_by = request.args.get("filter", default="participant_id,", type=str)
+    filter_by_col, filter_val = filter_by.split(",")[0], filter_by.split(",")[1]
+    filter_val = f"%{filter_val}%"
 
     try:
-        int(max_rows)
+        int(limit)
     except:
-        return "Max rows must be a valid integer", 400
+        return "Limit must be a valid integer", 400
+
+    try:
+        int(page)
+    except:
+        return "Page must be a valid integer", 400
+
+    offset = (int(page) * int(limit)) - int(limit)
 
     columns = models.Participant.__table__.columns.keys()
 
     if order_by_col not in columns:
         return f"Column name for ordering must be one of {columns}", 400
-    column = getattr(models.Participant, order_by_col)
+    if filter_by_col not in columns:
+        return f"Column name for filtering must be one of {columns}", 400
+
+    order_column = getattr(models.Participant, order_by_col)
+    filter_column = getattr(models.Participant, filter_by_col)
```

Comment:
This can be made more Pythonic by doing a try-except on getattr and catching the [AttributeError](https://docs.python.org/3/library/functions.html#getattr)

Reply:
fixed in 920f9b2c23911b11e012ee51312d3751d1142d33